crates/mirror-mirror/src/type_info/pretty_print.rs

```diff
@@ -0,0 +1,403 @@
+use core::fmt::{self, Write};
+
+use super::*;
+
+pub trait PrettyPrintRoot: super::private::Sealed {
+    fn pretty_print_root(&self) -> RootPrettyPrinter<'_, Self> {
+        RootPrettyPrinter { ty: self }
+    }
+
+    fn pretty_root_fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result;
+}
+
+#[derive(Debug)]
+pub struct RootPrettyPrinter<'a, T>
+where
+    T: ?Sized,
+{
+    ty: &'a T,
+}
+
+impl<'a, T> fmt::Display for RootPrettyPrinter<'a, T>
+where
+    T: PrettyPrintRoot,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.ty.pretty_root_fmt(f)
+    }
+}
+
+impl PrettyPrintRoot for TypeDescriptor {
+    fn pretty_root_fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.get_type().pretty_root_fmt(f)
+    }
+}
+
+impl<'a> PrettyPrintRoot for Type<'a> {
+    fn pretty_root_fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Type::Struct(inner) => inner.pretty_root_fmt(f),
+            Type::TupleStruct(inner) => inner.pretty_root_fmt(f),
+            Type::Tuple(inner) => inner.pretty_root_fmt(f),
+            Type::Enum(inner) => inner.pretty_root_fmt(f),
+            Type::List(inner) => inner.pretty_root_fmt(f),
+            Type::Array(inner) => inner.pretty_root_fmt(f),
+            Type::Map(inner) => inner.pretty_root_fmt(f),
+            Type::Scalar(inner) => inner.pretty_root_fmt(f),
+            Type::Opaque(inner) => inner.pretty_root_fmt(f),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+fn simple_type_name_fmt(type_name: &str, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    if let Some(name) = super::SimpleTypeName::new(type_name) {
+        write!(f, "{name}")
+    } else {
+        f.write_str(type_name)
+    }
+}
+
+#[cfg(not(feature = "std"))]
+fn simple_type_name_fmt(type_name: &str, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    f.write_str(type_name)
+}
+
+const IDENT: &str = "    ";
```

Comment:
```suggestion
const TAB: &str = "    ";
```
idk if `TAB` is actually best name here, but, something that signifies it's just alignment makes more sense than `IDENT` to me